File: tournaments/countSumOfTwoRepresentations3/countSumOfTwoRepresentations3.js
function countSumOfTwoRepresentations3(n, l, r) {
    let result = 0;
    let a = 1;

    while (n >= 2 * a) {
        if (l <= a && r >= n - a) ++result;
        ++a;
    }

    return result;
}

File: tournaments/countSumOfTwoRepresentations3/countSumOfTwoRepresentations3.cpp
int countSumOfTwoRepresentations3(int n, int l, int r) {
  int result = 0;

  for (int i = 1; i <= n - i; i++) {
    if (n - i <= r && l <= i) {
      result++;
    }
  }

  return result;
}
